rest/rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectorHandler.java

```diff
@@ -141,4 +153,44 @@ public Kind resourceKind() {
         final Map<String, String> props) {
         return verifier.verify(connectorId, encryptionComponent.decrypt(props));
     }
+
+    @GET
+    @Path("/{id}/icon")
+    public Response getConnectorIcon(@NotNull @PathParam("id") final String connectorId) {
+        Connector connector = get(connectorId);
+        String connectorIcon = connector.getIcon();
+
+        // If there is no specified icon, return 404
+        if (connectorIcon == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        // If the specified icon is a data URL, or a non-URL like value (e.g. font awesome class name), return 404
+        if (connectorIcon.startsWith("data:") || !connectorIcon.contains("/")) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        OkHttpClient httpClient = new OkHttpClient();
+        try {
+            final okhttp3.Response externalResponse = httpClient.newCall(new Request.Builder().get().url(connectorIcon).build()).execute();
```

Comment:
Who is supposed to server the icon ? Do we have an Icon service for this only or is this coming from somewhere externally (maybe the verifier service ?)

Reply:
The URL could be an external site, it could be the verifier, it could be a separate image icon service. I think leaving it like this gives flexibility.

A separate caching image service would be great though, so perhaps this is something we can extend to later. I need to work on the connector icon generator anyway so that could be refactored in that perhaps.

Reply:
Ok, but can we harden the external access at least a bit ? E.g. like setting a (small) timeout for the request ? And returning a default broken icon if the access fails ?